### PR TITLE
dependency: bump github.com/grpc-ecosystem/grpc-gateway/v2 from v2.15.0 to v2.17.1

### PR DESCRIPTION
**Pull request by aimuz**
Signed-off-by: aimuz <mr.imuz@gmail.com>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Closes #

**Special notes for your reviewer**:

ref: https://github.com/googleapis/go-genproto/issues/1015

### CODE REVIEW

**Comment by markmandel** [comment]
🤔 why does a dependency bump in released Agones affect any of the examples?

File: examples/autoscaler-webhook/go.mod [hunk 1]
```diff
@@ -47,4 +48,4 @@ require (
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
 
-replace agones.dev/agones => ../../
\ No newline at end of file
+replace agones.dev/agones => ../../
```
Comment (aimuz) [inline examples/autoscaler-webhook/go.mod:51]: @markmandel Here's a replace that should answer your question.

Comment (aimuz) [reply examples/autoscaler-webhook/go.mod:51]: replace will try to invalidate the versioning constraints of require, and the contents of replace will always prevail.

File: examples/simple-game-server/go.mod [hunk 2]
```diff
@@ -5,15 +5,17 @@ go 1.20
...
+	golang.org/x/sys v0.12.0 // indirect
+	golang.org/x/text v0.13.0 // indirect
+	google.golang.org/genproto v0.0.0-20230822172742-b8732ec3820d // indirect
+	google.golang.org/genproto/googleapis/api v0.0.0-20230822172742-b8732ec3820d // indirect
+	google.golang.org/genproto/googleapis/rpc v0.0.0-20230822172742-b8732ec3820d // indirect
+	google.golang.org/grpc v1.57.0 // indirect
+	google.golang.org/protobuf v1.31.0 // indirect
 )
 
 replace agones.dev/agones => ../../
```
Comment (aimuz) [inline examples/simple-game-server/go.mod:21]: And here, all instances of go.mod exist replace

File: sdks/swagger/sdk.swagger.json [hunk 3]
```diff
@@ -238,9 +238,6 @@
               "properties": {
                 "result": {
                   "$ref": "#/definitions/sdkGameServer"
-                },
```
Comment (markmandel) [inline sdks/swagger/sdk.swagger.json:241]: I wonder why this was here in the first place? 🤔

**Review by markmandel — APPROVED** [review]

File: sdks/swagger/sdk.swagger.json (hunk 3 above)
Comment (aimuz) [reply sdks/swagger/sdk.swagger.json:241]: I'll look into this.

Comment (aimuz) [reply sdks/swagger/sdk.swagger.json:241]: The option disable_default_errors exists, and it doesn't look like it should be there.

https://github.com/googleforgames/agones/blob/main/build/build-sdk-images/go/gen.sh#L56

**Pull request merged**